runelite-maven-plugin/src/main/java/net/runelite/mvn/ComponentMojo.java

```diff
@@ -70,11 +70,13 @@ public void execute() throws MojoExecutionException, MojoFailureException
 	{
 		TypeSpec.Builder interfaceType = TypeSpec.classBuilder("InterfaceID")
 			.addModifiers(Modifier.PUBLIC, Modifier.FINAL)
-			.addAnnotation(Deprecated.class);
+			.addAnnotation(Deprecated.class)
+			.addJavadoc("@deprecated Use {@link net.runelite.api.gameval.InterfaceID} instead");
 
 		TypeSpec.Builder componentType = TypeSpec.classBuilder("ComponentID")
 			.addModifiers(Modifier.PUBLIC, Modifier.FINAL)
-			.addAnnotation(Deprecated.class);
+			.addAnnotation(Deprecated.class)
+			.addJavadoc("@deprecated Use nested classes of {@link net.runelite.api.gameval.InterfaceID} instead");
```

Comment:
Wording suggestion
```suggestion
			.addJavadoc("@deprecated Use {@link net.runelite.api.gameval.InterfaceID} classes instead");
```

Reply:
I prefer mine since it makes it clear that InterfaceID itself is _not_ included in the alternatives.